beacon-chain/cache/depositsnapshot/deposit_inserter.go

```diff
@@ -28,6 +28,9 @@ var (
 func (c *Cache) InsertDeposit(ctx context.Context, d *ethpb.Deposit, blockNum uint64, index int64, depositRoot [32]byte) error {
 	ctx, span := trace.StartSpan(ctx, "Cache.InsertDeposit")
 	defer span.End()
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
```

Comment:
Why did you insert the error check here, but replaced `ctx` with `_` elsewhere?

Reply:
Good question... I started subjectively adding some context liveness checks but only did it in a few places. I can revert if it makes sense